### PR TITLE
Mh starting on UI

### PR DESCRIPTION
**Pull request by merrillhuang**
Merge starting on ui branch into main.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,3 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
```
Comment (aldoportillo) [inline app/controllers/application_controller.rb:2]: Good!

File: app/models/follow_request.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# Foreign Keys
+#
+#  fk_rails_...  (recipient_id => users.id)
+#  fk_rails_...  (sender_id => users.id)
+#
+class FollowRequest < ApplicationRecord
+  belongs_to :recipient, class_name: "User"
+  belongs_to :sender, class_name: "User"
+
+  enum status: { pending: "pending", rejected: "rejected", accepted: "accepted" }
```
Comment (aldoportillo) [inline app/models/follow_request.rb:26]: nice

File: app/models/photo.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+class Photo < ApplicationRecord
+  belongs_to :owner, class_name: "User", counter_cache: true
+  has_many :comments
+  has_many :likes
+  has_many :fans, through: :likes
+
+  validates :caption, presence: true
+  validates :image, presence: true
+
+  scope :past_week, -> { where(created_at: 1.week.ago...) }
```
Comment (aldoportillo) [inline app/models/photo.rb:31]: good scopes

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -8,9 +8,25 @@
 
     <%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>
     <%= javascript_importmap_tags %>
+
+    <%= render partial: "shared/cdn_assets" %>
   </head>
 
   <body>
-    <%= yield %>
```
Comment (aldoportillo) [inline app/views/layouts/application.html.erb:14]: good

**Review by aldoportillo — APPROVED** [review]

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (viviancreates) [reply app/controllers/application_controller.rb:2]: Great user authentication

File: app/models/follow_request.rb (hunk 2 above)
Comment (viviancreates) [reply app/models/follow_request.rb:26]: Good use of associations

File: app/models/photo.rb (hunk 3 above)
Comment (viviancreates) [reply app/models/photo.rb:31]: Great job with validations on caption and image

**Comment by viviancreates** [comment]
Looks good!

File: app/controllers/application_controller.rb
Comment (viviancreates) [inline app/controllers/application_controller.rb:1]: Great user authentication

File: app/models/follow_request.rb
Comment (viviancreates) [inline app/models/follow_request.rb:1]: Great user authentication

File: app/models/like.rb
Comment (viviancreates) [inline app/models/like.rb:1]: Good associations!

File: app/models/photo.rb
Comment (viviancreates) [inline app/models/photo.rb:1]: Great job with validations on caption and image

File: app/views/comments/index.html.erb
Comment (viviancreates) [inline app/views/comments/index.html.erb:1]: great links!

File: lib/tasks/dev.rake
Comment (viviancreates) [inline lib/tasks/dev.rake:1]: good job creating sample data

File: app/views/comments/_form.html.erb
Comment (viviancreates) [inline app/views/comments/_form.html.erb:1]: Nicely structured form!

**Pull request merged**